production/helm/loki/templates/_helpers.tpl

```diff
@@ -6,12 +6,63 @@ Expand the name of the chart.
 {{- coalesce .Values.nameOverride $default | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
+{{/*
+singleBinary fullname
+*/}}
```

Comment:
What do these helpers do?

Reply:
the way the `loki.name` and `loki.fullname` work between the existing `grafana/loki` and `grafana/loki-simple-scalable` charts differs a bit. To enable the migration path where the new chart will replace the stateful set created by `grafana/loki` we needed to bring in this function from `grafana/loki` to make sure they had the same name.

The helpers below `isSingleBinary` and `isScalable` determine which set of templates to deploy based on whether or not object storage was requested.